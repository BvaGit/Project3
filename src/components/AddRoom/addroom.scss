@import "../../utils/variables.scss";

$button-size: 40px;

.add-room {
  position: absolute;
  bottom: 0px;
  right: 0px;
  padding: 2px;
  font-size: $button-size;
  font-weight: bold;
  width: $button-size;
  height: $button-size;
  color: $background;
  border-radius: 50%;
  line-height: 33px;
  text-align: center;
  background: $button-color;
  margin: 0 $margin_md 20px;
  cursor: pointer;
}
.add-room:hover {
  background: $border;
  box-shadow: 0 14px 28px $button-color, 0 10px 10px $button-color;
  transition-duration: 0.3s;
}
