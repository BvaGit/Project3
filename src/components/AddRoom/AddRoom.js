import React from 'react';

import './addroom.scss';

const AddRoom = () => (
    <div className="add-room">
        +
    </div>
)

export default AddRoom;