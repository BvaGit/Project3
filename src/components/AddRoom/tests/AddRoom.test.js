import React from 'react';
import { shallow, mount } from 'enzyme';
import AddRoom from '../AddRoom';
import ReactDOM from 'react-dom';

jest.mock('react-dom',()=>({createPortal:jest.fn().mockReturnValue('<div></div>')}))

describe('AddRoom', () => {
    it("Should match snapshot", () => {
        const component = shallow(<AddRoom />)
        expect(component.html()).toMatchSnapshot()
    })
    it("Should", ()=>{
        const component = mount(<AddRoom />)
        console.log(component.debug())
        component.find(".add-room").simulate("click")
        expect(component.find("CustomModal")).toHaveLength(1)
    })
})