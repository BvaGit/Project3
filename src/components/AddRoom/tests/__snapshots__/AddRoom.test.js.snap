// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoom Should match snapshot 1`] = `"<div><div class=\\"add-room\\">+</div></div>"`;
